packages/eslint-plugin/src/rules/no-deprecated.ts

```diff
@@ -254,6 +288,10 @@ export default createRule<Options, MessageIds>({
         'Expected call like node to have signature',
       );
 
+      if (node.type === AST_NODE_TYPES.MemberExpression && node.computed) {
+        return getComputedPropertyDeprecation(node);
+      }
```

Comment:
[Refactor] `getComputedPropertyDeprecation` is already being called in `getDeprecationReason`, just after the `getCallLikeDeprecation`. If we swap the order to try the computed property check first then we'll be able to remove this call here:

```suggestion
```

```ts

    function getDeprecationReason(node: IdentifierLike): string | undefined {
      if (isInComputedProperty(node)) {
        return getComputedPropertyDeprecation(
          node.parent as TSESTree.MemberExpression,
        );
      }

      const callLikeNode = getCallLikeNode(node);
      if (callLikeNode) {
        return getCallLikeDeprecation(callLikeNode);
      }
```